mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/WorkerExecutionOperationsNetworkStage.java

```diff
@@ -622,7 +627,9 @@ private WorkerConsumer connectToObservableAtPreviousStages(Observable<JobSchedul
 
     @Override
     public void shutdownStage() throws IOException  {
-        logger.info("Shutdown initiated");
+        if (jobStatus != null) {
+            jobStatus.onNext(heartbeatRef.get().getFailedHeartbeatStatus());
```

Comment:
Wouldn’t it be cleaner to create the Status directly instead of altering the Heartbeat data structure?

Reply:
That's a fair ask. Let me explain why I implemented this in such way. 2 reasons mainly:
1. There is no reference to worker metadata in this class aside from within heartbeat instance.
2. I imagined heartbeat class to be more a status generator instead where heartbeat is "only a specific  type" but in such case we should probably rename the class.